File: pi.go
package main

import "fmt"
import "math/rand"
import "math"


const N_TRIALS = 100000
const N_BOUND = 10000000

func gcd(x int, y int) int{
    a := x
    b := y
    if b > a {
        b, a = x, y
    }
    for {
        if a == b {
            return a
        } else {
            x = a - b
            y = b

            if x > y {
                a, b = x, y
            } else {
                b, a = x, y
            }
        }

    }
}


func get_1_if_coprime(x int, y int) int {
    if gcd(x, y) == 1 {
        return 1
    } else{
        return 0
    }
}

func get_int() int {
    return 1 + rand.Intn(N_BOUND)
}



func calculate() float64{
    var coprimes int = 0
    var i int = 0
    var x int
    var y int
    var fraction_co_prime float64
    var pi_estimate float64
    for i <= N_TRIALS {
        i = i + 1
        x, y = get_int(), get_int()
        coprimes += get_1_if_coprime(x, y)
        if coprimes > 0{
            fraction_co_prime = float64(coprimes) / float64(i)
            pi_estimate = math.Sqrt(6 / fraction_co_prime)
        }
    }
    return pi_estimate

}


func main() {
    i := 0
    for i < 10 {
        i = i + 1
        fmt.Println(calculate())
    }
}

File: pi.py
import random
import math

N_TRIALS = 100000
N_BOUND = 10000000

# def gcd(a, b):
#     if a > b:
#         return gcd(b, a)
#     if a == b:
#         return a
#     return gcd(b - a, a)


def gcd(x, y):
    a, b = x, y
    if b > a:
        b, a = x, y

    while True:
        if a == b:
            return a
        x = a - b
        y = b

        if x > y:
            a, b = x, y
        else:
            b, a = x, y


def is_co_prime(a, b):
    if gcd(a, b) == 1:
        return 1
    return 0


def calculate():
    coprimes = 0
    for i in range(N_TRIALS):
        x, y = random.randint(1, N_BOUND), random.randint(1, N_BOUND)
        coprimes += is_co_prime(x, y)
        if coprimes:
            fraction_co_prime = coprimes / (i + 1.0)
            pi_estimate = math.sqrt(6 / fraction_co_prime)
    return pi_estimate


for i in range(10):
    estimate = calculate()
    print i, estimate
